tests/rptest/utils/si_utils.py

```diff
@@ -489,9 +489,20 @@ def kafka_start_offset(manifest) -> Optional[int]:
             return None
 
         start_model_offset = manifest['start_offset']
-        first_segment = min(manifest['segments'].values(),
-                            key=lambda seg: seg['base_offset'])
-        delta = first_segment['delta_offset']
+        segments_at_offset = list(
+            filter(lambda seg: seg['base_offset'] == start_model_offset,
+                   manifest['segments'].values()))
+        delta = -1
+        if len(segments_at_offset) == 0:
+            # If there's no segment that matches exactly, just use the first
+            # segment's delta, provided it's ahead of the start.
```

Comment:
I'm not too sure about this. Could you explain the rationale?

Reply:
I just did what I saw Redpanda doing, though admittedly this logic is wrong and should be using the segment's base offset rather than Redpanda's start offset.